### ISSUE TITLE
Incorrectly detecting language when checking multiple documents

### ISSUE THREAD

**Issue by Fjan** [validator/validator#463]
How to reproduce:
```vnu english.html``` => fine
```vnu french.html``` => fine
```vnu french.html english.html``` => 
"file:/english.html":1.16-2.16: info warning: This document appears to be written in French but the “html” start tag has “lang="en"”. Consider using “lang="fr"” (or variant) instead.

It looks like some variable does not get reset correctly between checking documents. However, I cannot reproduce the above problem consistently, sometimes it needs three or more files to trigger.

This does not happen with older versions of vnu, I suspect this commit is to blame: https://github.com/validator/validator/commit/d1ee5cdb941051625e29ce4ac6ca44b2f2a4b38a

**Comment by sideshowbarker** [comment]
Thanks for reporting this. If you can provide test documents with which I might be able to reproduce the problem (even intermittently), that would help much in trying to troubleshoot it.

**Comment by Fjan** [comment]
Ok, it consistently gives an error on the third document if you download these three pages as HTML and check them in one go.

https://www.supersaas.com/
https://www.supersaas.com/info/about
https://www.supersaas.fr/

**Comment by sideshowbarker** [comment]
Thanks for the test cases. I’m also able to consistently reproduce the bug as you describe.

> It looks like some variable does not get reset correctly between checking documents.

Yeah I would guess that’s probably what it is—or at least I hope it is, because otherwise it will be a lot harder to troubleshoot and fix.

Anyway I’ll investigate and see if I can find the cause.

**Comment by sideshowbarker** [comment]
I’ve found the cause and it turned out to be a thoughtless oversight in my code: There’s a buffer the language-detecting code uses to store the content of each document, and I had neglected to make the code clear that buffer after each document.

That oversight can cause a very bad unexpected effect in the case where someone gives the command-line checker multiple files. The more files you give it that are not all in the same language as one another, the worse the effect would be.

So thanks very much for reporting this. The fix will be in the next release later this month, and in the mean time you can get a jar with the fix at https://sideshowbarker.net/nightlies/jar/

**Comment by Fjan** [comment]
Thanks for the fix! Perhaps that might also explain the slowdown I was seeing when checking large directories.

**Comment by sideshowbarker** [comment]
> Perhaps that might also explain the slowdown I was seeing when checking large directories.

oh—yeah I guess it probably would, because I guess that buffer the checker is holding that document content in would grow until you run out of physical memory and start using swap
